hp-cli/src/main.rs

```diff
@@ -0,0 +1,160 @@
+    let output_folder = match output_folder {
+        Some(output_folder) => output_folder,
+        None => std::env::current_dir().unwrap(),
+    };
+
+    sd_card.push(DATA_FOLDER);
+
+    let count = WalkDir::new(sd_card.clone()).into_iter().count();
+    let progress_bar = ProgressBar::new(count as u64);
+
+    for entry in WalkDir::new(sd_card.clone()) {
+        let entry = entry?; 
+        let from = entry.path();
+        let to = output_folder.join(from.strip_prefix(sd_card.clone())?);
+
+        if entry.file_type().is_dir() {
+            fs::create_dir_all(to)?;
+        } else if entry.file_type().is_file() {
+            fs::copy(from, to)?;
+        }
+        progress_bar.inc(1);
+    }
+    progress_bar.finish();
+    Ok(())
+}
+
+
+pub fn merge_wavs(input: &std::path::PathBuf, output: &std::path::PathBuf) -> Result<(), Box<dyn Error>> {
+	// Read files from input directory
+	let mut files = std::fs::read_dir(input)?
+		.filter_map(|entry| entry.ok())
+		.filter(|entry| entry.file_type().ok().map(|t| t.is_file()).unwrap_or(false))
+		.filter(|entry| entry.path().extension().unwrap_or_default() == "wav")
+		.collect::<Vec<_>>();
+
+	// Sort files by name
+	files.sort_by(|a, b| a.file_name().cmp(&b.file_name()));
+
+	let output_name = files.first().unwrap().file_name();
+	let output_name = output_name.to_str().unwrap();
+
+	// Get wav spec from file
+	let spec = WavReader::open(files.first().unwrap().path())?.spec();
+	let mut writer = WavWriter::create(output.join(output_name), spec)?;
+
+    let progress_bar = ProgressBar::new(files.len() as u64);
+	match spec.sample_format {
+		hound::SampleFormat::Float => {
+			for file in files {
+				let mut reader = WavReader::open(file.path())?;
+				for sample in reader.samples::<f32>() {
+					writer.write_sample(sample?)?;
+				}
+                progress_bar.inc(1);
+			}
+		},
+		hound::SampleFormat::Int => {
+			for file in files {
+				let mut reader = WavReader::open(file.path())?;
+				for sample in reader.samples::<i32>() {
+					writer.write_sample(sample?)?;
+				}
+                progress_bar.inc(1);
+			}
+		},
+	}
+    progress_bar.finish();
+	writer.finalize()?;
+	Ok(())
+}
+
+fn main() {
+    let commands = Cli::parse();
+
+    match commands.commands {
+        Commands::Install(install) => {
+            println!("Installing hydrophonitor on SD card at {:?}", install.sd_card);
+        }
+        Commands::Import(mut import) => {
+            println!("Importing audio from SD card at {:?}", import.sd_card);
+
+            if let Some(output_folder) = &import.output_folder {
+                println!("Output folder: {:?}", output_folder);
+                import_from_sd(&mut import.sd_card, Some(output_folder.clone())).unwrap();
+            } else {
+                println!("Output folder: current directory");
+                import_from_sd(&mut import.sd_card, None).unwrap();
```

Comment:
Passing an invalid path to SD card causes a panic:

```
❯ cargo run -- import --sd-card /dev/sdb
    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
     Running `target/debug/hp-cli import --sd-card /dev/sdb`
Importing audio from SD card at "/dev/sdb"
Output folder: current directory
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { depth: 0, inner: Io { path: Some("/dev/sdb/home/pi/data"), err: Os { code: 20, kind: NotADirectory, message: "Not a directory" } } }', src/main.rs:136:59
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

This should be caught and user should be given an error message

Reply:
Addressed in https://github.com/nordic-dev-net/hydrophonitor/pull/14/commits/356e5db6b82f2cc4f3fb0001daaeb8072b3622ba